crates/modelardb_server/tests/integration_test.rs

```diff
@@ -168,6 +168,73 @@ fn test_can_get_schema() {
     stop_modelardbd(flight_server);
 }
 
+#[test]
+#[serial]
+fn test_can_list_actions() {
+    let temp_dir = tempfile::tempdir().expect("Could not create a directory.");
+    let flight_server = start_modelardbd(temp_dir.path());
+
+    let runtime = Runtime::new().expect("Unable to initialize runtime.");
+    let mut flight_service_client = create_apache_arrow_flight_service_client(&runtime, HOST, PORT)
+        .expect("Cannot connect to flight service client.");
+
+    let mut actions = runtime.block_on(async {
+        flight_service_client
+            .list_actions(Request::new(arrow_flight::Empty {}))
+            .await
+            .expect("Could not retrieve actions.")
+            .into_inner()
+            .map(|action| action.expect("Could not retrieve action.").r#type)
+            .collect::<Vec<String>>()
+            .await
+    });
+
+    // Sort() is called on the vector to ensure that the assertion will pass even if the order of
+    // the actions returned by the endpoint changes.
+    actions.sort();
+
+    assert_eq!(
+        actions,
+        vec![
+            "CollectMetrics",
+            "CommandStatementUpdate",
+            "FlushEdge",
+            "FlushMemory",
+        ]
+    );
+
+    stop_modelardbd(flight_server);
+}
+
+#[test]
+#[serial]
+fn test_can_collect_metrics() {
```

Comment:
It seems odd to only test the case where there are no metrics that have been recorded yet. The recording of metrics is itself tested in the storage engine but it would be relevant to test that this is correctly returned when calling the action. It can be a little difficult to force each of the metrics to have entries, specifically `used_compressed_memory` or `used_disk_space` but it should be possible to check that `used_uncompressed_memory` and `ingested_data_points` returns the correct metrics.

Reply:
Added work item in Azure for this.